src/ctapipe/containers.py

```diff
@@ -483,16 +483,12 @@ class DL1CameraContainer(Container):
 
     image = Field(
         None,
-        "Numpy array of camera image, after waveform extraction." "Shape: (n_pixel)",
-        dtype=np.float32,
```

Comment:
Since `ndim` is removed, the container complains when two gain images are filled:
```
ctapipe.core.container.FieldValidationError: DL1CameraContainer Field 'image': the value '[[76.30804  76.080025 76.71846  ... 84.10605  67.9628   84.71408 ]
 [65.03804  65.34207  74.57242  ... 82.70825  62.36258  73.0401  ]]' (<class 'numpy.ndarray'>) is invalid:  Should have numpy dtype float32
```

Reply:
Hmm. Perhaps we then need to break this into a parent contain and two childred that inherit from it, adding the `image`  field.  That way it's clear that DL1CameraContainer for Calibration events look different than calibrated ones.   Maybe DL1CameraContainer  (contains all common fields) → CalibratedDL1CameraContainer (contains float DL1 image with ndims=1), UncalibratedDL1CameraContainer (contains int dl1 image with ndims=2?)

Or something along those lines...

Reply:
What is the drawback of having it flexible as it is in the PR?
One problem with having two inheritance containers would be that if we want to store the calibration dl1 images for a single gain camera, we can not store them in `UncalibratedDL1CameraContainer` because of the dimension restrictions. Though, it would also not belong in `CalibratedDL1CameraContainer`. A semantic hardcoding structure, i.e. `OneGainDL1CameraContainer`; `TwoGainDL1CameraContainer`, would also not be ideal since then we don't have information if the `OneGainDL1CameraContainer` is gain_selected/calibrated or not.